### PR TITLE
python312Packages.setuptools: 70.0.0 -> 72.1.0

### PR DESCRIPTION
**Pull request by mweinelt**
https://setuptools.pypa.io/en/stable/history.html#v72-1-0

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/glyphtools/default.nix [hunk 1]
```diff
@@ -26,11 +25,9 @@ buildPythonPackage rec {
 
   # A unit test suite does exist, but it only contains a dummy test that
   # doesn't test anything. It does import the module though so we still run it.
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/glyphtools/default.nix:27]:
```suggestion
  # A unit test suite does exist, but it only contains a dummy test that
  # doesn't test anything.
```

File: pkgs/development/python-modules/pycdio/default.nix [hunk 2]
```diff
@@ -23,6 +23,8 @@ buildPythonPackage rec {
       --replace 'library_dirs=library_dirs' 'library_dirs=[dir.decode("utf-8") for dir in library_dirs]' \
       --replace 'include_dirs=include_dirs' 'include_dirs=[dir.decode("utf-8") for dir in include_dirs]' \
       --replace 'runtime_library_dirs=runtime_lib_dirs' 'runtime_library_dirs=[dir.decode("utf-8") for dir in runtime_lib_dirs]'
+    substituteInPlace test/test-cdtext.py \
+      --replace assertEquals assertEqual
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pycdio/default.nix:27]:
```suggestion
      --replace-fail 'library_dirs=library_dirs' 'library_dirs=[dir.decode("utf-8") for dir in library_dirs]' \
      --replace-fail 'include_dirs=include_dirs' 'include_dirs=[dir.decode("utf-8") for dir in include_dirs]' \
      --replace-fail 'runtime_library_dirs=runtime_lib_dirs' 'runtime_library_dirs=[dir.decode("utf-8") for dir in runtime_lib_dirs]'
    substituteInPlace test/test-cdtext.py \
      --replace-fail assertEquals assertEqual
```

**Review by danc86 — APPROVED** [review]

**Review by danc86 — APPROVED** [review]

**Comment by natsukium** [comment]
Does this PR turn into python-updates? I will also make time to work on it if necessary.

**Comment by mweinelt** [comment]
Eventually, but I only have time for a small run.

**Comment by mweinelt** [comment]
Now all that is left to do is wait for trofi to find out what else this change broke. :smile:

**Pull request merged**